src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java

```diff
@@ -90,88 +91,12 @@ public String getScheme() {
-     * provider allows a new file system to be created with the same URI as a
-     * file system it previously created.
-     *
-     * @param uri URI reference
-     * @param env A map of provider specific properties to configure the file system;
-     *            may be empty
-     * @return A new file system
-     *
-     * @throws FileSystemAlreadyExistsException if the file system has already been created
-     * @throws IllegalArgumentException if the pre-conditions for the uri parameter are not met, or the env parameter does not contain properties required by the provider, or a property value is invalid
+    /*
+     * @throws NotYetImplementedException
      */
     @Override
-    public S3FileSystem newFileSystem(URI uri, Map<String, ?> env)
-    throws FileSystemAlreadyExistsException {
-
-        if (uri == null) {
-            throw new IllegalArgumentException("uri can not be null");
-        }
-        if (uri.getScheme() == null) {
-            throw new IllegalArgumentException(
-                String.format("invalid uri '%s', please provide an uri as s3://bucket", uri.toString())
-            );
-        }
-        if (uri.getAuthority() == null) {
-            throw new IllegalArgumentException(
-                String.format("invalid uri '%s', please provide an uri as s3://bucket", uri.toString())
-            );
-        }
-
-        S3FileSystem fs = null;
-
-        S3FileSystemInfo info = fileSystemInfo(uri);
-        if (cache.containsKey(info.key())) {
-            throw new FileSystemAlreadyExistsException("a file system already exists for '" + info.key() + "', use getFileSystem() instead");
-        }
-
-        S3NioSpiConfiguration config = new S3NioSpiConfiguration(env)
-            .withEndpoint(info.endpoint())
-            .withBucketName(info.bucket())
-            ;
-
-        if (info.accessKey() != null) {
-            config.withCredentials(info.accessKey(), info.accessSecret());
-        }
-
-        cache.put(
-            info.key(),
-            fs = new S3FileSystem(this, config)
-        );
-
-        return fs;
-    }
-
-    /**
-     * Same as newFileSystem(uri, Collections.EMPTY_MAP);
-     *
-     * @param uri URI reference
-     *
-     * @return newFileSystem(uri, Collections.EMPTY_MAP)
-     *
-     * @throws FileSystemAlreadyExistsException if the file system has already been created
-     * @throws IllegalArgumentException if the pre-conditions for the uri parameter
-     *         are not met, or the env parameter does not contain properties
-     *         required by the provider, or a property value is invalid
-     */
-    public S3FileSystem newFileSystem(URI uri) {
-        return newFileSystem(uri, Collections.EMPTY_MAP);
+    public FileSystem newFileSystem(URI uri, Map<String, ?> env) {
+        throw new NotYetImplementedException("This method is not yet supported in v2.x. It might be implemented for bucket creation");
```

Comment:
Can you add an issue to implement this for bucket creation?

Reply:
Sure I'll do

Reply:
Created https://github.com/awslabs/aws-java-nio-spi-for-s3/issues/243, feel free to add any information you consider relevant 🙏